service/history/workflow/activity.go

```diff
@@ -57,7 +69,7 @@ func nextRetryDelayFrom(failure *failurepb.Failure) *time.Duration {
 	return delay
 }
 
-func updateActivityInfoForRetries(
+func UpdateActivityInfoForRetries(
```

Comment:
doesn't seem like you need to export this.

Reply:
Yeah. In my first implementation I was using it. Will revert this change.